### DISCUSSION TITLE
Slides position is not in the left margin on mounted

### DISCUSSION THREAD

**Discussion by leephan2k1** [davidjerleke/embla-carousel#378] [Help] [answered]
I followed the example: https://codesandbox.io/s/embla-carousel-slidestoscroll-react-forked-5lzuqp?file=/src/js/EmblaCarousel.js
But something is wrong when slides are initialized it has shifted to the right 582.6px

(I'm using tailwind, I think I switched the right style from css to tailwind)

![image](https://user-images.githubusercontent.com/54581258/192859089-0fa98fac-64d2-4b7a-ac77-ea3fcd6fc1d6.png)

I really want it to be in the left margin on initialization like this: translate3d(0px, 0px, 0px)

![image](https://user-images.githubusercontent.com/54581258/192860037-ad62157b-b4ac-4b2a-9c92-e02862f1e562.png)

**thanks for helping**

**Comment by davidjerleke** [comment, marked as answer]
Hi @leephan2k1,

Thank you for your question. I can’t debug if you don’t provide a CodeSandbox reproducing the problem. It could literally be anything causing this.

Let me know if you intend to provide a CodeSandbox.

Best,
David

**Reply by leephan2k1** [reply]
Sorry for late response. 
This is a CodeSandbox I tried to simulate my same problem
It always shifts to the right due to translate3d at initialization

https://codesandbox.io/s/embla-carousel-slidestoscroll-react-forked-2bonk2?file=/src/js/EmblaCarousel.js

**Reply by davidjerleke** [reply]
Hi @leephan2k1,

Thanks for the CodeSandbox. I found two problems:

1. If you want to clear leading and trailing empty space, add containScroll: trimSnaps:

```diff
const [viewportRef, embla] = useEmblaCarousel({
  slidesToScroll: 2,
  skipSnaps: false,
+ containScroll: "trimSnaps" // <-- add this
});
```

2. You're setting `width` on your slides but that's not how CSS flexbox works. If you want your slides to be `350px` wide you need to do this:

```diff
.embla__slide {
  position: relative;
- width: 350px; // <-- remove this
+ flex: 0 0 350px; // <-- add this
}
```

...or 

```diff
.embla__slide {
  position: relative;
  width: 350px;
+ flex: 0 0 auto; // <-- add this
}
```

I would suggest you to take some time to read the Embla documentation to learn how to setup your carousel, because these problems were caused by an incorrect setup:

- [Guide - Slide Container](https://www.embla-carousel.com/guides/slide-container/) 
- [Guide - Slide Sizes](https://www.embla-carousel.com/guides/slide-sizes/)
- [containScroll option](https://www.embla-carousel.com/api/options/#containscroll)

Best,
David

**Reply by leephan2k1** [reply]
Thanks a lot, I forgot to add containScroll option.

Sorry for the mistake in reading my document

**Reply by davidjerleke** [reply]
@leephan2k1 no worries. Just wanted to make you aware of the documentation which might be useful for you in other cases too.
